### DISCUSSION TITLE
Update form state in the parent component

### DISCUSSION THREAD

**Discussion by eliawk** [react-hook-form/react-hook-form#9810] [Q&A] [answered]
Hello!
I have a form parent component with thus state:
```
const [dataForm, setDataForm] = useState<any>({})
```
And I want yo update the state when the user interact with the form in the child component `<Form />`:

```
<Form
  onSubmit={formSubmit}
  defaultValues={defaultValues}
  setData={setDataForm}
  watchFields={['newPassword']}
  >
  <Input
  type='text'
  label='Old Password'
  name='oldPassword'
  validation={{
  required: 'The old password is required',
  minLength: {
	  value: 8,
	  message: 'the password must be at least 8 characters'
  }
  }}
  />
...
```
If I call `setData` in the Form component I get the Form component itself to re-render, so I get an infinite loop of rendering.
What is the best practice to pass the data to the parent component?

thanks

**Comment by bluebill1049** [comment]
not sure how your Form is built, perhaps you can use the new `values` prop form `useForm`

```tsx
useForm({
  values: dataForm
})
```

**Comment by eliawk** [comment, marked as answer]
hey @bluebill1049,
thanks for your answer and for the amazing library!
My problem was to send the data form the Form component to a parent component outside useForm.
I've resolved using this:
```
	useEffect(() => {
		const subscription = watch((data: any) => {
			console.log('watched data', data)
			setData(data)
		})
		return () => subscription.unsubscribe()
	}, [watch])
```
So now the setData is triggered just when the user interact in the form and update the outside parent controller.

**Comment by leapful** [comment]
@eliawk 

Based on your code example, in every re-rendering, the prop values are passing to `<Form />` component always be created as new reference values. Please memoize those props in order to prevent re-rendering. 

```
onSubmit={formSubmit}
defaultValues={defaultValues}
setData={setDataForm}
watchFields={['newPassword']}
```

If it does not work, please provide a codesandbox url to re-produce the issue.

**Reply by eliawk** [reply]
hey @leapful thanks for your advice. 
Atm I've fixed using the unsubscribe on watch method, but I think your approach will be useful in other cases as well!
